emails/utils.py

```diff
@@ -330,6 +330,9 @@ def remove_message_from_s3(bucket, object_key):
         )
         return response.get('DeleteMarker')
     except ClientError as e:
-        logger.error('s3_client_error_delete_email', extra=e.response['Error'])
+        if e.response['Error'].get('Code', '') == 'NoSuchKey':
+            logger.error('s3_object_does_not_exist', extra=e.response['Error'])
```

Comment:
I think we'll see this error twice  - once when we try to read the email from S3 in `_sns_response`, and once when try to delete the item. That's probably OK.

Reply:
I'll name this different errors so we can distinguish between the two.